src/components/icon/icon.md

```diff
@@ -0,0 +1,39 @@
+The `<Icon />` component's API is mostly self-evident and you can
+refer to the Props table at the bottom of this page. However, let's look
+at the color properties in more detail. While `fill`
+sets the color inside the object, and `stroke` sets
+color of the line drawn around the SVG, `currentColor` is used to,
+essentially, set the font `color` CSS property to the SVG icon. The
+reason it's called `currentColor` is that we utilize `currentColor`
+to allow an additional fill color to be applied by way of the fact that
+`currentColor` is a CSS value that tells the engine to use whatever CSS
+font `color` is inherited.
+
+The `currentColor` approach assumes the SVG _sub-elements_ of interest have
+something like `<path fill="currentColor" ...` defined within them. You
+can inspect the SVG itself to see this.
+
+Basic Usage
+
+```jsx
+const icon = require('../../svgs/icon-caution-fill.svg').default;
+<Icon name={icon.id} size="large" currentColor="caution" title="alert icon" />
+```
+Sizes
+
+```jsx
+const styles = require('./example.css'); /* For example purposes only! */
```

Comment:
Should we perhaps add a short `css` example to show how `currentColor` can be used?

Reply:
Gotcha, thanks!

Yeah, so, you cannot set this with CSS from the outside per se as the API will do it on your behalf from the icon component itself (via it's own CSS module in icon.css of course).

That said, I've added a small example to hopefully drill this in:

![image](https://user-images.githubusercontent.com/142403/62814776-75419000-bac7-11e9-819e-38c21cb30fd3.png)